TypeCobol/Compiler/Sql/CodeElements/SqlCodeElementBuilder.cs

```diff
@@ -349,5 +349,55 @@ public LockTableStatement CreateLockTableStatement(CodeElementsParser.LockTableS
 
             return new LockTableStatement(tableName, partitionId, mode);
         }
+
+        public DropTableStatement CreateDropTableStatement(CodeElementsParser.DropTableStatementContext context)
+        {
+            if (context.tableOrAliasName() != null)
+            {
+                var name = (Token) context.tableOrAliasName().tableOrViewOrCorrelationName().Name;
+                var schemaName = (Token) context.tableOrAliasName().tableOrViewOrCorrelationName().SchemaName;
+                var dbms = (Token) context.tableOrAliasName().tableOrViewOrCorrelationName().DBMS;
+                var tableOrAliasName = CreateTableOrAliasName(context.tableOrAliasName());
+                return new DropTableStatement(tableOrAliasName);
+            }
+            return null;
+        }
+
+        private TableViewCorrelationName CreateTableOrAliasName(CodeElementsParser.TableOrAliasNameContext context)
+        {
+            if (context.tableOrViewOrCorrelationName().Name == null) return null;
+            var nameToken = (Token) context.tableOrViewOrCorrelationName().Name;
+            var name = new SymbolReference(new AlphanumericValue(nameToken),
+                SymbolType.SqlIdentifier);
+            if (context.tableOrViewOrCorrelationName().SchemaName != null)
+            {
+                var qualifierToken = (Token) context.tableOrViewOrCorrelationName().SchemaName;
+                var qualifier =
+                    new SymbolReference(new AlphanumericValue(qualifierToken),
+                        SymbolType.SqlIdentifier);
+                if (context.tableOrViewOrCorrelationName().DBMS != null)
+                {
+                    var topLevelQualifierToken = (Token) context.tableOrViewOrCorrelationName().DBMS;
+                    var topLevelQualifier =
+                        new SymbolReference(new AlphanumericValue(topLevelQualifierToken),
+                            SymbolType.SqlIdentifier);
+                    var tail = new QualifiedSymbolReference(qualifier, topLevelQualifier);
+                    var fullName = new QualifiedSymbolReference(name, tail);
+                    return new TableViewCorrelationName(fullName);
+                }
+                else
+                {
+                    var fullName = new QualifiedSymbolReference(name, qualifier);
+                    return new TableViewCorrelationName(fullName);
+                }
+            }
+            else
+            {
+                var aliasName = new AmbiguousSymbolReference(new AlphanumericValue(nameToken),
+                    new SymbolType[] {SymbolType.SqlIdentifier, SymbolType.SqlIdentifier});
+                return new TableViewCorrelationName(aliasName);
+            }
+
+        }
```

Comment:
Ok But this can be simplified using the existing `CreateSymbolReference` method. If SchemaName is not null then you can safely call CreateSymbolReference and return its result.